### DISCUSSION TITLE
Plotting the OCV and terminal voltage on the same graph

### DISCUSSION THREAD

**Discussion by MatHunt** [pybamm-team/PyBaMM#2413] [Q&A] [answered]
This is probably more of a Python question than a pybamm question. I have computed the OCV curve from the data used using:
`x=parameter_values["Positive electrode OCP [V]"]
y=parameter_values["Negative electrode OCP [V]"]
def OCV(t):
    return x-y`

giving the OCV as a function of t(I think). I can't seem to plot this as I get an "Attribute error" whatever that is. I'm guessing I can't plot the function. How do I plot two things on one graph?

Can I let the OCV as a function of time perhaps? that would be good to compare the terminal voltage with.

**Comment by YannickNoelStephanKuhn** [comment]
You _did_ mean to say `return x(t) - y(t)`, right?

**Reply by MatHunt** [reply]
I've now changed it, and I still can't plot the thing.

**Comment by rtimms** [comment, marked as answer]
I think this does what you want, but the individual OCPs might not actually line up this way to give the OCV (e.g. fully-charged might not be sto=1, 0 in neg., pos.).

```python3
import pybamm
import matplotlib.pyplot as plt

parameter_values = pybamm.ParameterValues("Chen2020")

x = pybamm.linspace(0, 1, 1000)  # sto
U_n = parameter_values["Negative electrode OCP [V]"]
U_p = parameter_values["Positive electrode OCP [V]"]

fig, ax = plt.subplots(1, 3, figsize=(12, 4))
ax[0].plot(x.entries, U_n(x).entries)
ax[0].set(xlabel="sto [-]", ylabel="$U_n$ [V]")
ax[1].plot(x.entries, U_p(x).entries)
ax[1].set(xlabel="sto [-]", ylabel="$U_p$ [V]")
ax[2].plot(x.entries, U_p(x).entries - U_n(1-x).entries)
ax[2].set(xlabel="sto [-]", ylabel="$U$ [V]")
plt.tight_layout()
```

**Reply by MatHunt** [reply]
Although this works, I have no idea why, and I can't seem to figure out how it works.

**Reply by rtimms** [reply]
`x = pybamm.linspace(0, 1, 1000)` creates a `pybamm.Array` object. Doing `U_n = parameter_values["Negative electrode OCP [V]"]` loads the function from the parameters. You can call this by doing `U_n(x)` and the results is a `pybamm.Vector`. You can call `.entries` to return the elements of the `Vector` as an ordinary numpy array for plotting.

If you've already solved the model the easiest way to make the plot is to do
```python3
import pybamm

model = pybamm.lithium_ion.DFN()
parameter_values = pybamm.ParameterValues("Chen2020")
sim = pybamm.Simulation(model, parameter_values=parameter_values)
sim.solve([0, 3600])
sim.plot([["Terminal voltage [V]", "Measured open circuit voltage [V]"]])
```
